tests/integ/link_test.py

```diff
@@ -529,6 +546,133 @@ def testGet(self):
         self.assertEqual(link["title"], "slink")
         self.assertEqual(link["h5path"], "somevalue")
 
+    def testGetRecursive(self):
+        # test getting links from an existing domain, following links
+        domain = helper.getTestDomain("tall.h5")
+        print("testGetRecursive", domain)
+        headers = helper.getRequestHeaders(domain=domain)
+
+        # verify domain exists
+        req = helper.getEndpoint() + "/"
+        rsp = self.session.get(req, headers=headers)
+        if rsp.status_code != 200:
+            print(f"WARNING: Failed to get domain: {domain}. Is test data setup?")
+            return  # abort rest of test
+
+        rspJson = json.loads(rsp.text)
+        root_uuid = rspJson["root"]
+        self.assertTrue(root_uuid.startswith("g-"))
+
+        # get links for root group and other groups recursively
+        req = helper.getEndpoint() + "/groups/" + root_uuid + "/links"
+        params = {"follow_links": 1}
+        rsp = self.session.get(req, params=params, headers=headers)
+        self.assertEqual(rsp.status_code, 200)
+        rspJson = json.loads(rsp.text)
+        self.assertTrue("hrefs" in rspJson)
+        hrefs = rspJson["hrefs"]
+        self.assertEqual(len(hrefs), 3)
+        self.assertTrue("links" in rspJson)
+        grp_links = rspJson["links"]
+        hardlink_count = 0
+        softlink_count = 0
+        extlink_count = 0
+        expected_group_links = ("g1", "g2", "g1.1", "g1.2", "g1.2.1", )
+        expected_dset_links = ("dset1.1.1", "dset1.1.2", "dset2.1", "dset2.2")
+        expected_soft_links = ("slink", )
+        expected_external_links = ("extlink", )
+        self.assertEqual(len(grp_links), 6)
```

Comment:
Nitpick: It's not clear what this 6 corresponds to, since `expected_group_links` has 5 entries. 
Does `grp_links` refer to 'links under the targeted group in the request' rather than 'links to groups'? Renaming `grp_links` would make this more clear.